### PR TITLE
Fixes sprintf(): Too few arguments in form transformer

### PR DESCRIPTION
**Pull request by pedrocasado**
Similar to: #29482

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | none
| License       | MIT
| Doc PR        | none

Fixes the form reverse transformation when the method viewToNorm is called within a value with the character %:

Before : "sprintf(): Too few arguments"

After : Form reverse transformation works.

Reference : http://php.net/manual/function.sprintf.php

### CODE REVIEW

**Comment by xabbuh** [comment]
Can we add a test case to prevent future regressions (IIRC this was done recently to satisfy fabbot.io)?

File: src/Symfony/Component/Form/Form.php [hunk 1]
```diff
@@ -1112,7 +1112,7 @@ private function viewToNorm($value)
                 $value = $transformers[$i]->reverseTransform($value);
             }
         } catch (TransformationFailedException $exception) {
-            throw new TransformationFailedException(sprintf('Unable to reverse value for property path "%s": '.$exception->getMessage(), $this->getPropertyPath()), $exception->getCode(), $exception);
+            throw new TransformationFailedException(sprintf('Unable to reverse value for property path "%s": "%s".', $this->getPropertyPath(), $exception->getMessage()), $exception->getCode(), $exception);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Form.php:1115]: in order to remove the extra double quotes, the concatenation should be moved outside the sprintf

**Comment by ro0NL** [comment]
Hi @pedrocasado can you also have a look at #36954 maybe :)

File: src/Symfony/Component/Form/Form.php (hunk 1 above)
Comment (pedrocasado) [reply src/Symfony/Component/Form/Form.php]: fixed that. i see 3 more cases like this in the same file, but i don't know how i would reproduce the error. should i put the same fix?

Comment (stof) [reply src/Symfony/Component/Form/Form.php]: yes please.

Comment (stof) [reply src/Symfony/Component/Form/Form.php]: btw, we have this issue in other places too. This has been introduced in https://github.com/symfony/symfony/pull/36437

Comment (pedrocasado) [reply src/Symfony/Component/Form/Form.php]: @stof, i also fixed the files you pointed out there.

File: src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php [hunk 2]
```diff
@@ -126,14 +126,14 @@ protected function getConstructor(Definition $definition, $required)
...
-            throw new RuntimeException(sprintf('Invalid service "%s": '.lcfirst($e->getMessage()), $this->currentId));
+            throw new RuntimeException(sprintf('Invalid service "%s": ', $this->currentId).lcfirst($e->getMessage()));
         }
         if (!$r = $r->getConstructor()) {
             if ($required) {
                 throw new RuntimeException(sprintf('Invalid service "%s": class%s has no constructor.', $this->currentId, sprintf($class !== $this->currentId ? ' "%s"' : '', $class)));
             }
         } elseif (!$r->isPublic()) {
-            throw new RuntimeException(sprintf('Invalid service "%s": %s must be public.', $this->currentId, sprintf($class !== $this->currentId ? 'constructor of class "%s"' : 'its constructor', $class)));
+            throw new RuntimeException(sprintf('Invalid service "%s": "%s" must be public.', $this->currentId, sprintf($class !== $this->currentId ? 'constructor of class "%s"' : 'its constructor', $class)));
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php:136]: This case should be reverted. It is a false positive of fabbot (due to the nested `sprintf`)

Comment (pedrocasado) [reply src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php]: thats true. thanks

**Review by Tobion — CHANGES_REQUESTED** [review]
Seems like most (all?) of this was already fixed, e.g. in https://github.com/symfony/symfony/pull/36684
You need to rebase.

**Comment by Tobion** [comment]
Please add a test as suggested by @xabbuh for at least one case like the Form transformation.

**Comment by pedrocasado** [comment]
@Tobion , can you give me some guidance? do you think this test is enough?

```
public function testDataTransformationEscapesValueCorrectlyInException()
{
    $this->expectException('Symfony\Component\Form\Exception\TransformationFailedException');
    $this->expectExceptionMessage('Unable to transform data for property path "name": No mapping for value "90%"');
    $model = new FixedDataTransformer([
        'default' => 'foo',
    ]);
    $view = new FixedDataTransformer([
        'foo' => 'bar',
    ]);

    $config = new FormConfigBuilder('name', null, $this->dispatcher);
    $config->addViewTransformer($view);
    $config->addModelTransformer($model);
    $config->setData('90%');
    $form = new Form($config);

    $form->getData();
}
```

**Review by nicolas-grekas — APPROVED** [review]
(let's skip adding tests for this...)

**Comment by nicolas-grekas** [comment]
Thank you @pedrocasado.

**Pull request merged**